src/librustc/infer/mod.rs

```diff
@@ -1546,7 +1546,7 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
         Some(self.tcx.closure_kind(def_id))
     }
 
-    pub fn closure_type(&self, def_id: DefId) -> ty::PolyFnSig<'tcx> {
+    pub fn fn_sig(&self, def_id: DefId) -> ty::PolyFnSig<'tcx> {
```

Comment:
I believe someone also suggested `fn_sig_of` or just `sig_of`. But the latter doesn't fit with the type name.